src/prometheus_handler.go

```diff
@@ -67,56 +80,53 @@ func makePromHistogram(label string, histogram map[string]int, datatype string,
 		bounds = append(bounds, bound)
 	}
 	sort.Strings(bounds)
-	labelname = regexp.MustCompile(`[^a-zA-Z0-9 ]+`).ReplaceAllString(labelname, "")
-	if datatype=="1"{
-		datatype=labelname
+	MLName = regexp.MustCompile(`[^a-zA-Z0-9 ]+`).ReplaceAllString(MLName, "")
+	if MLValue == "1" {
+		MLValue = MLName
 	}
 	for _, bound := range bounds {
-		entry := fmt.Sprintf(`%s_bucket{le="%s", %s="%s"} %d`, label, bound, strings.TrimRight(labelname, "bt"), datatype, histogram[bound])
+		entry := fmt.Sprintf(`%s_bucket{le="%s", %s="%s"} %d`, label, bound, strings.TrimRight(MLName, "bt"), MLValue, histogram[bound])
 		output += "\n" + entry
 	}
 	entry := fmt.Sprintf("%s_count %d", label, histogram["+inf"])
 	output += "\n\n" + entry + "\n"
 	return output
 }
 
-func makePromGauge(label string, value string) string {
+func makePromGauge(label string, value string, MLName string, MLValue string) string {
+	var entry string
 	output := fmt.Sprintf(`
 # HELP %s gauge output
-# TYPE %s gauge\n`, label, label)
-	entry := fmt.Sprintf(`%s%s %s`, output, label, value)
+# TYPE %s gauge 
+`, label, label)
+	if len(MLName) > 0 {
+		entry = fmt.Sprintf(`%s%s{%s="%s"} %s`, output, label, MLName, MLValue, value)
+	} else {
+		entry = fmt.Sprintf(`%s%s %s`, output, label, value)
+	}
 	return entry + "\n"
 }
 
 func GenericPromDataParser(structure HandlerStructure) string {
 	var data string
-	for j := 0; j < len(structure); j++ {
+	for i := 0; i < len(structure); i++ {
 		{
 			var op string
-			//Reflect the struct
-			typeExtract := reflect.TypeOf(structure[j].Structure)
-			valueExtract := reflect.ValueOf(structure[j].Structure)
-
-			//Iterating over fields and compress their values
-			for i := 0; i < typeExtract.NumField(); i++ {
-				fieldType := typeExtract.Field(i)
-				fieldValue := valueExtract.Field(i)
-
-				promType := fieldType.Tag.Get("type")
-				promLabel := fieldType.Tag.Get("metric")
-				switch promType {
-				case "histogram":
-					histogram := parseHistogram(fieldValue)
-					op += makePromHistogram(promLabel, histogram, structure[j].Datatype, structure[j].Labelname)
-				case "counter":
-					count := parseCounter(fieldValue)
-					op += makePromCounter(promLabel, count)
-				case "gauge":
-					value := parseGauge(fieldValue)
-					op += makePromGauge(promLabel, value)
-				case "untype":
-					op += makePromUntype(promLabel, fieldValue)
-				}
+			fieldValue := reflect.ValueOf(structure[i].MValue)
```

Comment:
Can we add some comments here to explain the process happening here

Reply:
ha yes!